src/rules/scaled-access-relationships-claim.js

```diff
@@ -0,0 +1,124 @@
+*/
+function scaledAccessAddRelationshipsClaim(user, context, callback) {
+    const fetch = require("node-fetch");
+    const { URLSearchParams } = require('url');
+    const jwt = require('jsonwebtoken');
+
+    const getM2mToken = () => {
+        if (global.scaledAccessM2mToken && global.scaledAccessM2mTokenExpiryInMillis > new Date().getTime() + 60000) {
+            return Promise.resolve(global.scaledAccessM2mToken);
+        } else {
+            const tokenUrl = configuration.SCALED_ACCESS_TOKENURL || `https://${auth0.domain}/oauth/token`;
+            return fetch(tokenUrl, {
+                method: 'POST',
+                body: new URLSearchParams({
+                    grant_type: 'client_credentials',
+                    client_id: configuration.SCALED_ACCESS_CLIENTID,
+                    client_secret: configuration.SCALED_ACCESS_CLIENTSECRET,
+                    audience: configuration.SCALED_ACCESS_AUDIENCE,
+                    scope: 'pg:tenant:admin'
+                })
+            })
+                .then(response => {
+                    if (!response.ok) {
+                        return response.text().then((error) => {
+                            console.error("Failed to obtain m2m token from " + tokenUrl);
+                            throw Error(error);
+                        });
+                    } else {
+                        return response.json();
+                    }
+                })
+                .then(({ access_token }) => {
+                    global.scaledAccessM2mToken = access_token;
+                    global.scaledAccessM2mTokenExpiryInMillis = jwt.decode(access_token).exp * 1000;
+                    return access_token;
+                });
+        }
+    };
+
+    const callRelationshipManagementApi = async (accessToken, path) => {
+        const url = `${configuration.SCALED_ACCESS_BASEURL}/${configuration.SCALED_ACCESS_TENANT}/${path}`;
+        return fetch(url, {
+            method: 'GET',
+            headers: {
+                "Authorization": "Bearer " + accessToken,
+                "Content-Type": "application/json"
+            }
+        })
+            .then(async response => {
+                if (response.status === 404) {
+                    return [];
+                } else if (!response.ok) {
+                    return response.text().then((error) => {
+                        console.error("Failed to call relationship management API", url);
+                        throw Error(error);
+                    });
+                } else {
+                    return response.json();
+                }
+            });
+    };
+
+    const getRelationships = (accessToken) => {
+        return callRelationshipManagementApi(accessToken, `actors/user/${user.user_id}/relationships`);
+    };
+
+    const addClaimToToken = (apiResponse) => {
+        const claimName = configuration.SCALED_ACCESS_CUSTOMCLAIM || `https://${auth0.domain}/relationships`;
```

Comment:
The `domain` here will fail as anything with `auth0.com` cannot be used. 

https://auth0.com/docs/tokens/create-namespaced-custom-claims

Reply:
Oh. Changed to `https://scaledaccess.com/relationships`